src/DataProtection/DataProtection/src/TypeForwardingActivator.cs

```diff
@@ -64,6 +62,27 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
         }
 
         protected string RemoveVersionFromAssemblyName(string forwardedTypeName)
-            => _versionPattern.Replace(forwardedTypeName, "");
+        {
+            // Type, Assembly, Version={Version}, Culture={Culture}, PublicKeyToken={Token}
+
+            var versionStartIndex = forwardedTypeName.IndexOf("Version=", StringComparison.Ordinal);
+            while (versionStartIndex != -1)
```

Comment:
Why is this a while loop?

Reply:
The test case that fails is the type is generic and there's multiple Version strings...  So I made it a while loop to reg ex replace all instances...:shrug: